### PR TITLE
[SDPA] Remove the chunk_grad from mem-eff attention

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

There exists an optimization within the scaled_dot_product_efficieint bacwkard attention path to, under the right conditions, output grad_q, grad_k, grad_v all as aliases of the same storage. This was done to optimize for the hot path where mha does packed linear_projection -> chunk -> (view stuff) -> sdpa. The thought was that chunk-> would be able to "trivially" cat inputs to chunk.backward(). However upon closer inspection chunk.backward will call ` cat` irregardless of the inputs so this is not being utilized. 

I validated this by profiling on main and then this branch and the traces produced the same both with `split.backward()` calling into cat.

### CODE REVIEW

**Comment by drisspg** [comment]
Traced through chunk_backward and it is calls `torch.cat` irregardless if the input tensorlist share storage memory. Potential optimization there. SO not sure if I should maybe try and make that optimzation or remove the chunk_grad like I did in this PR.

**Review by cpuhrsch — APPROVED** [review]

**Comment by danthe3rd** [comment]
> However upon closer inspection chunk.backward will call  cat irregardless of the inputs so this is not being utilized.

This is something that could be improved in PyTorch. In xFormers we have a custom [`chunk`](https://github.com/facebookresearch/xformers/blob/main/xformers/ops/unbind.py#L112) operator, which for its backward will `cat` inputs, either by calling `cat`, or by changing the gradient view if they are already in contiguous regions of memory.
This gives significative performance improvement for smaller sequence lengths (like for ViTs, which use 196/256 seqlen).

**Comment by drisspg** [comment]
@albanD and I spoke about this and hand some concerns potential edge cases. IIRC: There isn't a way to denote that the functions may be aliases of each other and not of the input. And there were concerns about the about this pattern from a correctness perspective with arbitrary backward graphs
